diesel_migrations/migrations_internals/src/lib.rs

```diff
@@ -200,6 +200,55 @@ where
     Ok(pending)
 }
 
+pub fn revert_all_migrations_in_directory<Conn>(
+    conn: &Conn,
+    path: &Path,
+) -> Result<Vec<String>, RunMigrationsError>
+where
+    Conn: MigrationConnection,
+{
+    let number = migrations_in_directory(path)?.len() as i64;
+
+    revert_latest_migrations_in_directory(conn, path, number)
+}
+
+// TODO : see if we need a function named revert_latest_migrations
+// to call this one.
+pub fn revert_latest_migrations_in_directory<Conn>(
+    conn: &Conn,
+    path: &Path,
+    number: i64,
+) -> Result<Vec<String>, RunMigrationsError>
+where
+    Conn: MigrationConnection,
+{
+    setup_database(conn)?;
+
+    let latest_migration_versions = match conn.latest_run_migration_versions(number) {
+        Ok(migration_versions) => migration_versions,
+        Err(_) => {
+            return Err(RunMigrationsError::MigrationError(
+                MigrationError::NoMigrationRun,
+            ))
+        }
+    };
+
+    if latest_migration_versions.is_empty() {
+        return Err(RunMigrationsError::MigrationError(
+            MigrationError::NoMigrationRun,
+        ));
+    }
+
+    for migration_version in &latest_migration_versions {
+        // TODO see how to handle the errors... even if these errors
```

Comment:
I think the correct behavior here is to return the error and stop reverting anything at this point. We cannot be sure that the next migrations will be fine and will not damage anything.